plugins/inputs/prometheus/kubernetes.go

```diff
@@ -126,6 +150,169 @@ func (p *Prometheus) watch(ctx context.Context, client *k8s.Client) error {
 	}
 }
 
+func (p *Prometheus) cAdvisor(ctx context.Context, client *k8s.Client) error {
+	p.Log.Infof("Using monitor pods version 2 to get pod list using cAdvisor.")
+
+	nodeIP := os.Getenv("NODE_IP")
+	if nodeIP == "" {
+		return errors.New("The environment variable NODE_IP is not set. Cannot get pod list for monitor_kubernetes_pods using version 2.")
+	}
+
+	// Parse label and field selectors - will be used to filter pods after cAdvisor call
+	labelSelector, err := labels.Parse(p.KubernetesLabelSelector)
+	if err != nil {
+		p.Log.Errorf("Error parsing the specified label selector(s): %s", err.Error())
+	}
+	fieldSelector, err := fields.ParseSelector(p.KubernetesFieldSelector)
+	if err != nil {
+		p.Log.Errorf("Error parsing the specified field selector(s): %s", err.Error())
+	}
+	if fieldSelectorIsSupported(fieldSelector) {
+		p.Log.Errorf("A specified field selector is not supported as a field selector for pods")
+	}
+	p.Log.Infof("Using the label selector: %v and field selector: %v", labelSelector, fieldSelector)
+
+	// Set InsecureSkipVerify for cAdvisor client since Node IP will not be a SAN for the CA cert
+	tlsConfig := client.Client.Transport.(*http.Transport).TLSClientConfig
+	tlsConfig.InsecureSkipVerify = true
+
+	// The request will be the same each time
+	podsUrl := fmt.Sprintf("https://%s:10250/pods", nodeIP)
+	req, err := http.NewRequest("GET", podsUrl, nil)
+	if err != nil {
+		return err
+	}
+	client.SetHeaders(req.Header)
+
+	// Update right away so code is not waiting the length of the cAdvisorCallInterval initially
+	err = updateCadvisorPodList(ctx, p, client, req, labelSelector, fieldSelector)
+	if err != nil {
+		return err
```

Comment:
Consider wrapping more context in certain places where you think it might be helpful to aid in debugging problems. for example:
```
		return fmt.Errorf("updating pod list: %w", err)
```